test/cpp/end2end/client_lb_end2end_test.cc

```diff
@@ -634,8 +635,11 @@ TEST_F(ClientLbEnd2endTest, ChannelStateConnectingWhenResolving) {
   // Note that this call also returns IDLE, since the state change has
   // not yet occurred; it just gets triggered by this call.
   EXPECT_EQ(channel->GetState(true /* try_to_connect */), GRPC_CHANNEL_IDLE);
-  // Now that the channel is trying to connect, we should be in state
+  // Now that the channel is trying to connect, we should get to state
   // CONNECTING.
+  while (channel->GetState(false) == GRPC_CHANNEL_IDLE) {
```

Comment:
Instead of a sleep loop, this can use:

```
ASSERT_TRUE(WaitForChannelState(
    channel.get(),
    [&](grpc_connectivity_state state) {
      if (state == GRPC_CHANNEL_IDLE) return false;
      EXPECT_EQ(state, GRPC_CHANNEL_CONNECTING);
      return true;
    }));
```

Reply:
done